Inconsistent formatting since clang-format 9 merge

**Describe the bug**
Found files with inconsistent formatting: 
```
MSGBLD0220: + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +
MSGBLD0220: +                 STATIC CODE ANALYSIS DETECTED PROBLEMS !                    +
MSGBLD0220: + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +

MSGBLD0220: C/C++ files with formatting errors:
MSGBLD0220: ... libnestutil/block_vector.h
MSGBLD0220: ... libnestutil/enum_bitfield.h
MSGBLD0220: ... libnestutil/lockptr.h
MSGBLD0220: ... libnestutil/sort.h
MSGBLD0220: ... nestkernel/connection_creator.cpp
MSGBLD0220: ... nestkernel/connection_manager.cpp
MSGBLD0220: ... nestkernel/event.h
MSGBLD0220: ... nestkernel/grid_layer.h
MSGBLD0220: ... nestkernel/nest_time.h
MSGBLD0220: ... nestkernel/node_collection.h
MSGBLD0220: ... nestkernel/ntree.h
MSGBLD0220: ... nestkernel/position.h
MSGBLD0220: ... nestkernel/stimulation_backend_mpi.cpp
MSGBLD0220: ... nestkernel/target_data.h
MSGBLD0220: ... sli/dict.cc
MSGBLD0220: ... sli/dict.h
MSGBLD0220: ... sli/tarrayobj.h
MSGBLD0220: ... sli/token.h
MSGBLD0220: ... sli/tokenarray.h
Error: Process completed with exit code 1.
```

**To Reproduce**
Steps to reproduce the behavior:
checkout to master and run clang-format on any of the files above. 
in terminal: `diff -u nestkernel/node_collection.h /dev/stdin <<<$(clang-format nestkernel/node_collection.h)`

**Expected behavior**
No difference when running clang-format on the (untouched) files above.

**Desktop/Environment (please complete the following information):**
 - OS Ubuntu 20.04
 - Python-Version: 3.9
 - NEST-Version: master
 - Installation: nest-simulator conda environment; apt-installed clang-format version 10 (although this is the wrong version, formatting for example nestkernel/node_collection.h with clang-format 10 fixed the issue for this file.)

**Additional context**
see static tests failing here: https://github.com/mlober/nest-simulator/runs/4941813778?check_suite_focus=true

After fixing the formatting of one of the files, the static checks on github ran perfectly, even though the other files were still broken --> The testsuit seems to only check for changes (?)

--- a/build_support/ci_build.sh
+++ b/build_support/ci_build.sh
@@ -94,7 +94,7 @@ if [ "$xNEST_BUILD_TYPE" = "STATIC_CODE_ANALYSIS" ]; then
     # The names of the static code analysis tools executables.
     VERA=vera++
     CPPCHECK=cppcheck
-    CLANG_FORMAT=clang-format
+    CLANG_FORMAT=clang-format-9
     PEP8=pycodestyle
     PYCODESTYLE_IGNORES="E121,E123,E126,E226,E24,E704,W503,W504"
 

